core-plugins/src/main/java/io/cdap/plugin/batch/action/EmailAction.java

```diff
@@ -94,11 +100,24 @@ public PasswordAuthentication getPasswordAuthentication() {
       javaMailProperties.put("mail.smtp.ssl.enable", true);
     }
     if ("TLS".equalsIgnoreCase(config.protocol)) {
-      javaMailProperties.put("mail.smtp.starttls.enable", "true");
-      javaMailProperties.put("mail.smtp.ssl.trust", config.host);
+      javaMailProperties.put("mail.smtp.starttls.enable", true);
+      if (!userJavaMailProperties.containsKey("mail.smtp.starttls.enable")) {
+        javaMailProperties.put("mail.smtp.starttls.enable", "true");
+      }
+      if (!userJavaMailProperties.containsKey("mail.smtp.ssl.trust")) {
+        javaMailProperties.put("mail.smtp.ssl.trust", config.host);
+      }
       protocolForTransport = "smtp"; // protocol is smtp not tls.
     }
 
+     if (!userJavaMailProperties.containsKey("mail.smtps.ssl.protocols")) {
+       javaMailProperties.put("mail.smtps.ssl.protocols", "TLSv1.2");
```

Comment:
Discussed offline, but we might also need to set these configurations for `mail.smtp.ssl.protocols` for use in startTLS as well, thanks!

Reply:
yeah I am running that test rn.